src/irmin-pack/pack_index.ml

```diff
@@ -35,9 +35,10 @@ module Make (K : Irmin.Hash.S) = struct
   module Key = struct
     type t = K.t [@@deriving irmin]
 
-    let hash = Irmin.Type.(unstage (short_hash K.t)) ?seed:None
+    let hash t =
+      Irmin.Type.(unstage (short_hash K.t)) ?seed:None t |> ( land ) 0x7fffffff
```

Comment:
You should hoist `unstage` here to avoid allocations due to generic specialisation in the hot path.

Reply:
Ha yes good catch: @pascutto can you move the `unstage` function out of the closure, so it's allocated only once?